tests/src/Loader/binding/tracing/BinderEventListener.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using System.Threading;
+using System.Reflection;
+using Xunit;
+
+using Assert = Xunit.Assert;
+
+namespace BinderTracingTests
+{
+    internal class BindEvent
+    {
+        internal AssemblyName AssemblyName;
+        internal bool Success;
+
+        internal Guid ActivityId;
+        internal Guid ParentActivityId;
+
+        internal bool Completed;
+        internal bool Nested;
+    }
+
+    internal sealed class BinderEventListener : EventListener
+    {
+        public Dictionary<Guid, BindEvent> BindEvents = new Dictionary<Guid, BindEvent>();
+
+        private const EventKeywords TasksFlowActivityIds = (EventKeywords)0x80;
+        private const EventKeywords BinderKeyword = (EventKeywords)0x4;
+
+        private object eventsLock = new object();
```

Comment:
Newbie question: can't we lock on BindEvents instead?

Reply:
That is a good question. In general, taking a lock on a `public` object instance is to be discouraged since anyone could technical take a lock on it.  If the `BindEvents` field was marked `private`, then yes that would make sense.

Reply:
I guess `BindEvents` could actually be `private` here. That said, I'm generally in the habit of having a dedicated object for the lock, as I think it is easy to miss updating the lock if something previously `private` were later exposed.

Reply:
>I'm generally in the habit of having a dedicated object for the lock

That is a decent policy, however the counter-point is avoiding allocations on performance critical paths. Remember that extra allocations should always be avoided in performance critical scenarios (i.e. start up). In this case I don't believe that is an issue, but it should be considered when applying the "dedicated object for the lock" policy.

Reply:
Agreed (hence generally). This is definitely not a path where performance is critical.